packages/next-auth/src/core/lib/cookie.ts

```diff
@@ -121,6 +121,18 @@ export interface Cookie extends CookieOption {
 
 type Chunks = Record<string, string>
 
+/**
+ * Sort cookies by chunk name to avoid joining them in the wrong order
+ */
+function sortedObject(obj) {
```

Comment:
Probably wouldn't be an issue, but I'm not sure this would work correctly if there where more than 10 chunks, where the order would be 0,1,10,2,3,4... using this function.